From d8b363f93682b8545ecdd9b30f8b8de352783eb1 Mon Sep 17 00:00:00 2001
From: "Teng, Jin Chung" <jin.chung.teng@intel.com>
Date: Fri, 22 Oct 2021 17:50:08 +0800
Subject: [PATCH] Force disable MMC and Scalability as read from user registry 
 will fail.

Force disable MMC and scalability on will caused GPU Hang during multi channel decode.

Signed-off-by: Teng, Jin Chung <jin.chung.teng@intel.com>
---
 media_driver/linux/common/os/mos_os_specific.c    | 3 +++
 media_driver/linux/gen12/ddi/media_sku_wa_g12.cpp | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/media_driver/linux/common/os/mos_os_specific.c b/media_driver/linux/common/os/mos_os_specific.c
index 610f2aeb..9257540a 100755
--- a/media_driver/linux/common/os/mos_os_specific.c
+++ b/media_driver/linux/common/os/mos_os_specific.c
@@ -7303,6 +7303,9 @@ static MOS_STATUS Mos_Specific_InitInterface_Ve(
             osInterface->bHcpDecScalabilityMode = MOS_SCALABILITY_ENABLE_MODE_USER_FORCE;
         }
 
+	//WA: Force Scalability to turn off as it will caused GPU Hang
+        osInterface->bHcpDecScalabilityMode = MOS_SCALABILITY_ENABLE_MODE_FALSE;
+
 #if (_DEBUG || _RELEASE_INTERNAL)
         osInterface->frameSplit                  = false;
         MOS_ZeroMemory(&userFeatureData, sizeof(userFeatureData));
diff --git a/media_driver/linux/gen12/ddi/media_sku_wa_g12.cpp b/media_driver/linux/gen12/ddi/media_sku_wa_g12.cpp
index a4179c86..5a49389d 100644
--- a/media_driver/linux/gen12/ddi/media_sku_wa_g12.cpp
+++ b/media_driver/linux/gen12/ddi/media_sku_wa_g12.cpp
@@ -266,6 +266,9 @@ static bool InitTglMediaSku(struct GfxDeviceInfo *devInfo,
         MEDIA_WR_SKU(skuTable, FtrE2ECompression, 0);
     }
 
+    // WA: Force MMC turn off as it will cause GPU Hang during multi channel decode
+    MEDIA_WR_SKU(skuTable, FtrE2ECompression, 0);
+
     // Force MMC Turn on for all components if set reg key
     MOS_ZeroMemory(&userFeatureData, sizeof(userFeatureData));
     MOS_UserFeature_ReadValue_ID(
-- 
2.25.1

